desktop/cmp/input/CodeInput.js

```diff
@@ -179,28 +209,125 @@ export class CodeInput extends HoistInput {
                     ref: this.manageCodeEditor,
                     onChange: this.onChange
                 }),
-                this.renderActionButtons()
+                this.enableSearch ? this.renderToolbar() : this.renderActionButtons()
             ],
-
             className: this.getClassName(),
             onBlur: this.onBlur,
             onFocus: this.onFocus,
-
             ...props
         });
     }
 
+    renderToolbar() {
+        const {editor, match, matches, showCopyButton, showFormatButton, showFullscreenButton,
+            formModel, hasFocus, fullScreen, showToolbar} = this;
+        if (!hasFocus) return null;
+
+        if (!showToolbar) {
+            return this.renderActionButtons();
+        } else {
+            return hbox({
+                className: 'xh-code-input__toolbar',
+                item: toolbar({
+                    items: [
+                        form({
+                            model: formModel,
+                            items: [
+                                formField({
+                                    field: 'query',
+                                    label: null,
+                                    item: textInput({
+                                        leftIcon: Icon.search(),
+                                        width: 150,
+                                        enableClear: true,
+                                        rightElement: hbox(
+                                            label({
+                                                className: 'xh-code-input--label',
```

Comment:
We are definitely not perfect at following the BEM CSS classname convention throughout the code - not by any means! - but where we can it's nice.  Here I would use `xh-code-input__label` since it's a nested block element, not a modifier (like e.g. `--selected` would be)